Applications/spire/source/time_and_sales/time_and_sales_window.cpp

```diff
@@ -1,16 +1,23 @@
 #include "spire/time_and_sales/time_and_sales_window.hpp"
+#include <QKeyEvent>
+#include <QVBoxLayout>
 #include "spire/time_and_sales/empty_time_and_sales_model.hpp"
+#include "spire/security_input/security_input_dialog.hpp"
+#include "spire/security_input/security_input_model.hpp"
 
 using namespace boost;
 using namespace boost::signals2;
 using namespace Nexus;
 using namespace spire;
 
 time_and_sales_window::time_and_sales_window(
-    const time_and_sales_properties& properties, QWidget* parent)
-    : QWidget(parent) {
+    const time_and_sales_properties& properties, 
+    security_input_model& input_model, QWidget* parent)
+    : QWidget(parent),
+      m_input_model(&input_model) {
   set_properties(properties);
   set_model(std::make_shared<empty_time_and_sales_model>(Security()));
+  setWindowTitle(tr("New Window"));
```

Comment:
The title window should be as follows:

If the security is invalid, then the window title is "Time and Sales" otherwise for a security with ticker symbol XXX.YYY it's "XXX.YYY - Time and Sales"